index.bs

```diff
@@ -11302,6 +11306,174 @@ The [=remote end steps=] given |session| and |command parameters| are:
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+WebExtensionsCommand = (
+  webExtensions.Install,
+  webExtensions.Uninstall
+)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+WebExtensionsResult = (
+  webExtensions.InstallResult
+)
+</pre>
+
+### Types ### {#module-webExtensions-types}
+
+#### The webExtensions.Extension Type #### {#type-webExtensions-Extension}
+
+<pre class="cddl remote-cddl local-cddl">
+webExtensions.Extension = text
+</pre>
+
+The <code>webExtensions.Extension</code> type represents a web extension id within a browser.
+
+### Commands ### {#module-webExtensions-commands}
+
+#### The webExtensions.install Command #### {#command-extensions-install}
+
+The <dfn export for=commands>webExtensions.install</dfn> command installs a web extension for the session.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      webExtensions.Install = (
+         method: "webExtensions.install",
+         params: webExtensions.InstallParameters
+      )
+
+      webExtensions.InstallParameters = {
+         extensionData: webExtensions.ExtensionData,
+      }
+
+      webExtensions.ExtensionData = (
+         webExtensions.ExtensionPath /
+         webExtensions.ExtensionArchivePath /
+         webExtensions.ExtensionBase64Encoded
+      )
+
+      webExtensions.ExtensionPath = {
+         type: "path",
+         path: text,
+      }
+
+      webExtensions.ExtensionArchivePath = {
+         type: "archivePath",
+         path: text,
+      }
+
+      webExtensions.ExtensionBase64Encoded = {
+         type: "base64",
+         value: text,
+      }
+      </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+      webExtensions.InstallResult = {
+        extension: webExtensions.Extension
+      }
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for webExtensions.install">
+The [=remote end steps=] with |command parameters| are:
+
+1. If installing web extensions isn't supported return [=error=] with error code [=unsupported operation=].
+
+1. Let |extension data spec| be a value of <code>extensionData</code> field of the |command parameters|.
+
+1. Let |extension directory entry| be the result of [=trying=] to [=expand a web extension data spec=] with |extension data spec|.
+
+1. Perform implementation defined steps to install a web extension from the path specified by |extension directory entry|. If this fails, return [=error=] with [=error code=] [=invalid web extension=]. Otherwise let |extension id| be the unique identifier of the newly installed web extension.
+
+1. Let |result| be a [=/map=] matching the
+   <code>webExtensions.InstallResult</code> production with the
+   <code>extension</code> field set to |extension id|.
+
+1. Return [=success=] with data |result|.
+
+</div>
+
+<div algorithm>
+To <dfn>expand a web extension data spec</dfn> given |extension data spec|:
```

Comment:
Question: Compared to other commands we have the remote end steps following at the end and other algorithms coming first. I think that we should follow that style.